core/src/main/codegen/templates/Parser.jj

```diff
@@ -2186,18 +2186,47 @@ SqlNode TableRef3(ExprContext exprContext, boolean lateral) :
     final SqlIdentifier alias;
     final Span s;
     SqlNodeList args;
+    final List<SqlNode> tableFuncArgs;
     final SqlNodeList columnAliasList;
     SqlUnnestOperator unnestOp = SqlStdOperatorTable.UNNEST;
 }
 {
     (
         LOOKAHEAD(2)
-        tableName = CompoundTableIdentifier()
-        ( tableRef = TableHints(tableName) | { tableRef = tableName; } )
-        [ tableRef = ExtendTable(tableRef) ]
-        tableRef = Over(tableRef)
-        [ tableRef = Snapshot(tableRef) ]
-        [ tableRef = MatchRecognize(tableRef) ]
+        tableName = CompoundTableIdentifier() { s = span(); }
+        (
+            // Three tokens needed to disambiguate EXTEND syntax from CALCITE-493.
+            // Example: "FROM EventLog(lastGCTime TIME)".
+            LOOKAHEAD(3)
+            // Table call syntax like FROM a.b() instead of FROM TABLE(a.b())
+            <LPAREN>
+            {  tableFuncArgs = new ArrayList<SqlNode>(); }
+            [
+                AddArg0(tableFuncArgs, ExprContext.ACCEPT_CURSOR)
+                (
+                    <COMMA> {
+                        // a comma-list can't appear where only a query is expected
+                        checkNonQueryExpression(ExprContext.ACCEPT_CURSOR);
+                    }
+                    AddArg(tableFuncArgs, ExprContext.ACCEPT_CURSOR)
+                )*
+            ]
+            <RPAREN>
+            {
+                final SqlParserPos pos = s.end(this);
+                tableRef = createCall(tableName, pos,
```

Comment:
here you initialize `tableRef` by creating a call, and then the next line you reassign its value by wrapping it in another call, i was curious why you did that. to keep the code simple by breaking across two lines? should it be two separate variables?

Reply:
I was following the convention of all the other code in the grammar, and the code in this production in particular.

It's really common to have this accumulator pattern:

```
    sqlNode = ParseSomething()
    sqlNode = ParseSomethingElse(sqlNode)
    sqlNode = ParseAThirdThing(sqlNode)
```

So I followed the same pattern in the construction of the SqlNode tree here.

But I don't care strongly about it.